src/main/scala/ch/openolitor/stammdaten/StammdatenUpdateService.scala

```diff
@@ -180,6 +184,49 @@ class StammdatenUpdateService(override val sysConfig: SystemConfig) extends Even
     }
   }
 
+  def updateZusatzabosWithNewName(meta: EventMetadata, newName: String, id: AbotypId)(implicit personId: PersonId = meta.originator) = {
+    DB localTxPostPublish { implicit session => implicit publisher =>
+      val zusatzabos = stammdatenWriteRepository.getZusatzAbosByZusatzabotyp(id)
+      zusatzabos map { zusatzabo =>
+        val copy = copyFrom(zusatzabo, zusatzabo, "abotypName" -> newName, "modifidat" -> meta.timestamp, "modifikator" -> personId)
+        stammdatenWriteRepository.updateEntityFully[ZusatzAbo, AboId](copy)
+      }
+    }
+  }
+
+  def updateMainAbosWithNewZusatzabotypName(meta: EventMetadata, oldName: String, newName: String, id: AbotypId)(implicit personId: PersonId = meta.originator) = {
+    DB localTxPostPublish { implicit session => implicit publisher =>
+      val zusatzabos = stammdatenWriteRepository.getZusatzAbosByZusatzabotyp(id)
+      zusatzabos map { zusatzabo =>
+
+        val mainAbos = stammdatenWriteRepository.getDepotAbosByZusatzAboId(zusatzabo.id) ++
+          stammdatenWriteRepository.getPostAbosByZusatzAboId(zusatzabo.id) ++
+          stammdatenWriteRepository.getHeimAbosByZusatzAboId(zusatzabo.id)
```

Comment:
We usually made this aggregation in the repository.

Reply:
done